methods/array_ops/src/pg_gp/array_ops.sql_in

```diff
@@ -602,11 +602,22 @@ m4_ifdef(`__HAS_FUNCTION_PROPERTIES__', `NO SQL', `');
  *        in later GPDB and Postgres versions.
  */
 DROP AGGREGATE IF EXISTS MADLIB_SCHEMA.array_agg(anyelement) CASCADE;
-CREATE AGGREGATE MADLIB_SCHEMA.array_agg( anyelement) (
-   SFUNC     = array_append,
-   STYPE     = anyarray
-   m4_ifdef( `__POSTGRESQL__', `', `, PREFUNC   = array_cat')
-);
+DO $$
+BEGIN
+  --
+  -- array_agg() compatibility function is only needed PostgreSQL 9.4 or earlier
+  -- because 9.5 or later has pg_catalog.array_agg(anyarray).
+  --
+  IF regexp_replace(version(), '.*PostgreSQL (\d+\.\d+).*', '\1')::float8 < 9.5 THEN
```

Comment:
I'm confused about this change and why we even need this function. Older versions of GPDB did not have array_agg (since they were based on PG 8.3) and needed this function. All currently supported platforms have this functionality. I realize 9.5 also supports array_agg(anyarray), but this function was not intended to provide that feature.